data/scenarios/Challenges/ice-cream.yaml

```diff
@@ -0,0 +1,242 @@
+version: 1
+name: Ice Cream Shop
+author: Karl Ostmo
+description: |
+  Serve a demanding customer
+creative: false
+seed: 0
+objectives:
+  - goal:
+      - |
+        Congratulations on the grand opening of your new ice cream shop.
+        You have advertised: "All you can eat, for 1 bitcoin!"
+      - |
+        A customer is approaching the "Counter". They look hungry!
+      - "..."
+      - |
+        Customer: "`give` me a cone, and then I'll tell you how many scoops I want."
+      - "..."
+      - |
+        Oh dear, you've forgotten to stock your shop with a "calculator".
+        Let's hope this order is simple.
+    condition: |
+      customer <- robotnamed "customer";
+      as customer {has "cone"};
+  - goal:
+      - |
+        `give` the customer exactly what they request.
+      - |
+        Perhaps there is something else around the shop that can assist with
+        preparing the order...
+    condition: |
+      as base {
+        has "bitcoin";
+      };
+robots:
+  - name: base
+    dir: [0, 1]
+    devices:
+      - branch predictor
+      - treads
+      - antenna
+      - comparator
+      - workbench
+      - grabber
+      - dictionary
+      - lodestone
+      - lambda
+      - logger
+      - welder
+      - hearing aid
+      - net
+      - strange loop
+    inventory:
+      - [5000, egg]
+      - [5000, milk]
+      - [5000, sugar]
+      - [5000, ice]
+      - [5000, vanilla]
+  - name: customer
+    system: true
+    dir: [0, -1]
+    display:
+      invisible: false
+      attr: 'iron'
+    inventory:
+      - [1, bitcoin]
+    program: |
+      run "scenarios/Challenges/_ice-cream/customer.sw"
+solution: |
+   run "scenarios/Challenges/_ice-cream/solution.sw"
+entities:
+  - name: scoop
+    display:
+      char: '@'
+    description:
+      - A single scoop of vanilla ice cream
+    properties: [portable]
+  - name: cherry
+    display:
+      char: '6'
+      attr: 'red'
+    properties: [portable]
+    description:
+      - A traditional ice cream garnish
+  - name: Counter
+    display:
+      char: '▒'
+      attr: 'sand'
+    properties: [portable, known]
+    capabilities: [count]
+    description:
+      - Where customers are served.
+      - Also... it can count things.
```

Comment:
:laughing:  Unfortunately I never noticed the `Counter`.  I just manually wrote a function `x2288` which repeats its argument 2288 times (by defining `x64 = \c. x32 c; x32 c` and so on).

Reply:
Suggestion: what about equipping the base with a `scanner`?  It was a bit annoying wandering around trying to `grab` everything to see what it was.

Reply:
@kostmo I would suggest that the customer wait outside while we prepare the order.

That way the counter will be visible. :slightly_smiling_face:

Reply:
> Suggestion: what about equipping the base with a `scanner`? It was a bit annoying wandering around trying to `grab` everything to see what it was.

To be fair, it's best that you grab all of those items anyway :)